src/ClientResolver.php

```diff
@@ -252,7 +252,9 @@ public function resolve(array $args, HandlerList $list)
                 if (isset($a['default'])) {
                     // Merge defaults in when not present.
                     $args[$key] = is_callable($a['default'])
-                    && !is_string($a['default'])
+                    && (in_array('callable', $a['valid'])
```

Comment:
Let's break this out from a ternary expression into some more readable like if statements.